src/adapter/vertex.rs

```diff
@@ -36,31 +36,34 @@ pub enum VertexKind<'a> {
     ImplementedTrait(&'a Path, &'a Item),
     FunctionParameter(&'a str),
     FunctionAbi(&'a Abi),
+    Discriminant(String),
+    Variant(EnumVariant<'a>),
 }
 
 impl<'a> Typename for Vertex<'a> {
     /// The name of the actual runtime type of this vertex,
     /// intended to fulfill resolution requests for the __typename property.
     #[inline]
     fn typename(&self) -> &'static str {
-        match self.kind {
+        match &self.kind {
             VertexKind::Item(item) => match &item.inner {
                 rustdoc_types::ItemEnum::Module { .. } => "Module",
                 rustdoc_types::ItemEnum::Struct(..) => "Struct",
                 rustdoc_types::ItemEnum::Enum(..) => "Enum",
                 rustdoc_types::ItemEnum::Union(..) => "Union",
                 rustdoc_types::ItemEnum::Function(..) => "Function",
-                rustdoc_types::ItemEnum::Variant(variant) => match variant.kind {
-                    VariantKind::Plain => "PlainVariant",
-                    VariantKind::Tuple(..) => "TupleVariant",
-                    VariantKind::Struct { .. } => "StructVariant",
-                },
                 rustdoc_types::ItemEnum::StructField(..) => "StructField",
                 rustdoc_types::ItemEnum::Impl(..) => "Impl",
                 rustdoc_types::ItemEnum::Trait(..) => "Trait",
                 rustdoc_types::ItemEnum::Constant(..) => "Constant",
                 rustdoc_types::ItemEnum::Static(..) => "Static",
                 rustdoc_types::ItemEnum::AssocType { .. } => "AssociatedType",
+                // TODO: How are we even here???
```

Comment:
Was this a note-to-self, or something you wanted me to answer?

Reply:
Both. I was having a really hard time understanding how I was ending up in this branch just reading `edges.rs`.

Reply:
This query will get you there:
```graphql
{
    Crate {
        item {
            ... on Enum {
                variant {
                    __typename @output
                }
            }
        }
    }
}
```

If the enum has all three kinds of variants, all three branches of the inner `match` will get hit too.

Reply:
Also, remember that `match` arms are evaluated in order, so it's usually a good idea to not reorder them unless you've measured perf and determined the change is an improvement.